src/QuickMarcEditor/QuickMarcEditorRows/FixedField/FixedFieldFactory.js

```diff
@@ -1,64 +1,112 @@
+} from '../BytesField';
 
-import BookFixedField from './BookFixedField';
-import ComputerFileFixedField from './ComputerFileFixedField';
-import ContinuingResourceFixedField from './ContinuingResourceFixedField';
-import MapFixedField from './MapFixedField';
-import MixedMaterialFixedField from './MixedMaterialFixedField';
-import ScoreFixedField from './ScoreFixedField';
-import SoundRecordingFixedField from './SoundRecordingFixedField';
-import VisualMaterialFixedField from './VisualMaterialFixedField';
 import HoldingsFixedField from './HoldingsFixedField';
 import AuthorityFixedField from './AuthorityFixedField';
+import BibliographicFixedField from './BibliographicFixedField';
 
 export const FixedFieldFactory = {
-  getFixedFieldByType(marcType, type, subtype) {
+  getFixedFieldByType(marcType) {
     let FixedField;
 
-    switch (true) {
-      case marcType === MARC_TYPES.HOLDINGS:
+    switch (marcType) {
+      case MARC_TYPES.HOLDINGS:
         FixedField = HoldingsFixedField;
         break;
-      case marcType === MARC_TYPES.AUTHORITY:
+      case MARC_TYPES.AUTHORITY:
         FixedField = AuthorityFixedField;
         break;
-      case type === 'a' && ['b', 'i', 's'].includes(subtype):
-      case type === 's':
-        FixedField = ContinuingResourceFixedField;
-        break;
-      case type === 'a' && ['a', 'c', 'd', 'm'].includes(subtype):
-      case type === 't':
-        FixedField = BookFixedField;
-        break;
-      case type === 'm':
-        FixedField = ComputerFileFixedField;
-        break;
-      case ['e', 'f'].includes(type):
-        FixedField = MapFixedField;
-        break;
-      case type === 'p':
-        FixedField = MixedMaterialFixedField;
-        break;
-      case ['c', 'd'].includes(type):
-        FixedField = ScoreFixedField;
-        break;
-      case ['i', 'j'].includes(type):
-        FixedField = SoundRecordingFixedField;
-        break;
-      case ['g', 'k', 'o', 'r'].includes(type):
-        FixedField = VisualMaterialFixedField;
+      case MARC_TYPES.BIB:
+        FixedField = BibliographicFixedField;
         break;
       default:
         FixedField = null;
     }
 
     return FixedField;
   },
-  getFixedField(name, marcType, type, subtype) {
-    const FixedField = this.getFixedFieldByType(marcType, type, subtype);
 
-    return FixedField ? <FixedField name={name} /> : null;
+  getDocumentType(marcSpec, type, subtype) {
+    let idx = -1;
+
+    if (!marcSpec?.spec) {
+      return undefined;
+    }
+
+    const checkTypeSubType = (orElement) => {
+      const hasTypePos = orElement.positions['6'] || false;
+      const hasSubTypePos = orElement.positions['7'] || false;
+      const typeOK = hasTypePos && orElement.positions['6'].includes(type);
+      const subTypeOK = hasSubTypePos ? hasSubTypePos && orElement.positions['7'].includes(subtype) : true;
+
+      return (typeOK && subTypeOK);
+    };
+
+    marcSpec.spec.types.forEach((marcType, indexType) => {
```

Comment:
I'd prefer to avoid indexes and use `find` method instead

```

const fixedFieldType = marcSpec.spec.types.find((fixedFieldType) => {
  return fixedFieldRype.identifiedBy.or.some((fieldIdentifier) => {
    return fieldIdentifier.tag === LEADER_TAG && checkTypeSubType(fieldIdentifier);
  });
});

return fixedFieldType;
```

Reply:
Nice and clean - ok